x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_tables/inline_tags_list.tsx

```diff
@@ -14,33 +14,39 @@ import { Query } from '../../types';
 
 interface Props {
   tags?: Query['tags'];
+  displayCountOnly?: boolean;
 }
-export const InlineTagsList: React.FC<Props> = ({ tags }) => {
+export const InlineTagsList: React.FC<Props> = ({ displayCountOnly, tags }) => {
   if (!tags?.length) return null;
 
   const displayedTags = tags.slice(0, 2);
-  const tooltipTags = tags.slice(2);
+  const tooltipTags = tags.slice(displayCountOnly ? 0 : 2);
+  const tagsText = tags.length === 1 ? 'tag' : 'tags';
+  const moreTagsMessage = displayCountOnly
+    ? `${tooltipTags.length} ${tagsText}`
+    : `and ${tooltipTags.length} more`;
 
-  return (
+  const TagToolTip = () => (
+    <EuiToolTip position="bottom" content={tooltipTags.join(', ')}>
+      <EuiBadge color={displayCountOnly ? 'hollow' : 'default'}>
+        {i18n.translate('xpack.enterpriseSearch.appSearch.engine.analytics.table.moreTagsBadge', {
+          defaultMessage: moreTagsMessage,
+          values: { moreTagsCount: tooltipTags.length },
+        })}
+      </EuiBadge>
+    </EuiToolTip>
+  );
+
+  const TagList = () => (
     <EuiBadgeGroup>
       {displayedTags.map((tag: string) => (
         <EuiBadge color="hollow" key={tag}>
           {tag}
         </EuiBadge>
       ))}
-      {tooltipTags.length > 0 && (
-        <EuiToolTip position="bottom" content={tooltipTags.join(', ')}>
-          <EuiBadge>
-            {i18n.translate(
-              'xpack.enterpriseSearch.appSearch.engine.analytics.table.moreTagsBadge',
-              {
-                defaultMessage: 'and {moreTagsCount} more',
-                values: { moreTagsCount: tooltipTags.length },
-              }
-            )}
-          </EuiBadge>
-        </EuiToolTip>
-      )}
+      {tooltipTags.length > 0 && <TagToolTip />}
     </EuiBadgeGroup>
   );
+
+  return displayCountOnly ? <TagToolTip /> : <TagList />;
```

Comment:
This change introduces a way to render _only_ a tooltip with a total tag count. Works well for cases where the tables are small and compact.

Reply:
I'm a little confused by this because it looks like _none_ of the tables are rendering a `<TagToolTip />` now, even though it feels like the RecentQueriesTable and the QueryClicksTable could:

<img width="1125" alt="" src="https://user-images.githubusercontent.com/549407/109574421-b7b13000-7aa4-11eb-80b8-4f4a41a6382a.png">
<img width="918" alt="" src="https://user-images.githubusercontent.com/549407/109574420-b5e76c80-7aa4-11eb-8b49-a31eca909c7c.png">

 From the code I think I see why and what we need to change, but just want to check with you first that we even still even want the `<TagToolTip />` component - it would be much simpler code-wise to only have one way of displaying tags.

Reply:
This should be fixed now in https://github.com/elastic/kibana/pull/92939/commits/5e1fc00cbe99501d544745a78beb9b295d5b7074:

![tags](https://user-images.githubusercontent.com/549407/109742201-82c7da80-7b83-11eb-9688-a64b41bc830e.gif)